### ISSUE TITLE
Sections break when not array in >2.7 (only in Node)

### ISSUE THREAD

**Issue by sup3rman** [linkedin/dustjs#671]
I havent had time to figure out exactly where the error is as dust fails without errors and never reaches the callback.

The following code breaks in Node, but seems to work in the broswer. If i downgrade to 2.6.2 i have no errors.

```
{#organization}
<div vocab="http://schema.org" typeof="Organization">
	<p property="legalName">{legalName}</p>
	<p typeof="PostalAddress" property="address">
		<span property="streetAddress">{address.streetAddress}</span><br />
		<span property="postalCode">{address.postalCode}</span> <span property="addressLocality">{address.addressLocality}</span>
	</p>

	<p>
		Epost: <a property="email" href="mailto:info@anderssonmotorsport.nu">{email}</a><br />
		Telefon: <span property="telephone">{telephone}</span>
	</p>
</div>
{/organization}
```

EDIT: If I turn organization into an array of organizations, it works immediately.

**Comment by sethkinast** [comment]
What node version? Can you share a representative context that fails?

I can't quite imagine this being broken as it's one of the most tested code paths, and we test in node, phantom, and all major browsers.

**Comment by sethkinast** [comment]
I tested with this code:

```
var dust = require('dustjs-linkedin');

var tmpl = dust.compile("{#organization}Name: {name} Address: {address.streetAddress}{/organization}", "hello");
dust.loadSource(tmpl);

dust.render("hello", { organization: { name: "Foo", address: { streetAddress: "123 Bar St" }} }, function(err, out) {
  if(err) {
    console.error(err);
  } else {
    console.log(out);
  }
});
```

And ran it using Node 0.8.28 and io.js 1.4.3. Each time, the output was as expected. Please provide any other info you can.

**Comment by sup3rman** [comment]
Hey again,

So yeah, took me a while to find what causes it... but it seems to be when I section ({#section}) a single NOT lean mongoose model. If I pass an array of NOT lean mongoose models it works.

If I make the mongoose model lean, then it works sectioning a single one.

The EXTREMELY annoying part was that Dust just completely freezes without outputting any errors, or calling the callback.

**Comment by sethkinast** [comment]
Ah ok, this is fixed in https://github.com/linkedin/dustjs/issues/663 and will be in Dust 2.7.2.

**Comment by sup3rman** [comment]
Shit sorry
Den 29 maj 2015 18:55 skrev "Seth Kinast" <notifications@github.com>:

**Comment by sethkinast** [comment]
Oh you have nothing to be sorry for, I wouldn't expect you to search closed issues, not your fault!

We will try to have a new release out very soon; sorry you wasted time trying to debug.

**Comment by sethkinast** [comment]
This should be resolved in 2.7.2
